File: example/1d-linear-convection/weno3/julia/03b/src/flux/factory.jl
# src/flux/factory.jl

module FluxCalculatorFactory

# 全局注册表
const _REGISTRY = Dict{String, Function}()

function register(name::String, ctor::Function)
    if haskey(_REGISTRY, name)
        @warn "通量计算器 '$name' 已注册，将被覆盖"
    end
    _REGISTRY[name] = ctor
end

function create(cfd::Any)
    flux_type = cfd.config.flux_type
    if !haskey(_REGISTRY, flux_type)
        error("未注册的通量计算器: '$flux_type'。可用: $(keys(_REGISTRY))")
    end
    return _REGISTRY[flux_type](cfd)
end

register("rusanov", cfd -> Main.RusanovFluxCalculator(cfd))
register("engquist-osher", cfd -> Main.EngquistOsherFluxCalculator(cfd))

end # module FluxCalculatorFactory

File: example/1d-linear-convection/weno3/python-v1/02c/src/flux/factory.py
# flux/factory.py
"""
通量计算器专用工厂（封装注册细节，提供清晰接口）
符合你希望“将创建逻辑封装在工厂中”的设计原则
"""

from core.registry import BaseFactory

class FluxCalculatorFactory:
    """通量计算器工厂：隐藏注册类别和组件名称等细节"""

    @staticmethod
    def create(cfd) -> 'InviscidFluxCalculator':
        """
        创建通量计算器实例
        
        Args:
            cfd: CFD 主对象，包含 config.flux_type 等配置
            
        Returns:
            实现 InviscidFluxCalculator 接口的通量计算器实例
        """
        flux_type = cfd.config.flux_type
        return BaseFactory.create_component('flux', flux_type, cfd)
   